src/Lib/Core/Jwt.hs

```diff
@@ -49,8 +49,8 @@ mkJWTToken (Seconds expiry) payload = do
   }
   return $ JWT.encodeSigned JWT.HS256 secret claimsSet
 
-decodeAndVerifyJWTToken :: (MonadIO m, MonadReader AppEnv m) => Text -> m (Maybe JWTPayload)
-decodeAndVerifyJWTToken token = do
+decodeAndVerifyJwtToken :: (MonadIO m, MonadReader AppEnv m) => Text -> m (Maybe JwtPayload)
```

Comment:
This function should take `JwtToken` as an argument instead of `Text`.

Note, that `JwtSecret /= JwtToken`. You need to have three `newtype`s here in total:

1. `JwtPayload`.
2. `JwtSecret`.
3. `JwtToken`.